fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryTypeFilterPlan.java

```diff
@@ -165,4 +171,35 @@ public void logPlanStructure(StoreTimer timer) {
     public int getComplexity() {
         return 1 + getInner().getComplexity();
     }
+
+    @Nonnull
+    @Override
+    public InternalPlannerGraph rewriteInternalPlannerGraph(@Nonnull List<InternalPlannerGraph> childGraphs) {
+        final InternalPlannerGraph.Node root =
+                new InternalPlannerGraph.Node(this,
+                        getClass().getSimpleName(),
+                        "[" + String.join(" v ", recordTypes) + "]");
+        final InternalPlannerGraph graphForInner =
+                Iterables.getOnlyElement(childGraphs);
+        return InternalPlannerGraph.builder(root)
+                .addGraph(graphForInner)
+                .addEdge(graphForInner.getRoot(), root, new InternalPlannerGraph.GroupExpressionRefEdge())
+                .build();
+    }
+
+    @Nonnull
+    @Override
+    public ExplainPlannerGraph rewriteExplainPlannerGraph(@Nonnull List<ExplainPlannerGraph> childGraphs) {
+        final ExplainPlannerGraph.Node root =
+                new ExplainPlannerGraph.Node(this,
+                        getClass().getSimpleName(),
+                        "[" + String.join(" v ", recordTypes) + "]");
```

Comment:
The rewrites with fancy string interpolation stuff are the ones that make me saddest about the code duplication.

Reply:
see above -- no reason to be sad anymore